mvc/models/model-binding.rst

```diff
@@ -1,8 +1,113 @@
+
+In the example, the model binder is loos through the Request.Form first for a key named `id`. In this case it didn't find anything, so it moved on to RouteData.Values looking for a key with a name of `id`. In our example, it's a match, since the last portion of the route has a key value of `id`, and it is an integer. Binding happens, and the value of the id parameter becomes 2. If the action method's parameter were a complex type such as the ``Movie`` type, which likely contains both primitive and complex types, MVC's model binder will still handle it nicely. It uses reflection and recursion to traverse the properties of complex types through its tree, and comparing any primitive members' names against the names of keys in data sources. The model binder can bind objects that contain nested objects for multiple levels. 
+
+.. note:: When the binder finds the first matching name in the data source, it stops looking for more elements matching that name, and moves onto the next property or item in the parameter list. 
+ 
+.. note:: If for some reason binding can't happen, it quietly fails.
+ 
+Additionally, there are some special data types that ASP.NET MVC must consider when performing model binding:
+
+- ``Request.Files``: Uploaded files that are part of the HTTP request.
+- ``IFormCollection``: The collection of key-value pairs that represent the form data.
+- Cancelation token: Used to cancel activity in asynchronous controllers.
+
+Once model binding is complete, validation occurs. The default model binders are quite smart, and work great for the vast majority of development scenarios. ASP.NET MVC is extensible so if you have unique needs you can customize the built in model binders.  
+
+Customize default model binding behavior with attributes 
+--------------------------------------------------------
+ASP.NET MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source, which is what the ``[From*]`` attributes do, such as the ``[FromForm]`` or ``[FromQuery]``. While the topic of model binding extensibility in depth is outside of the scope of this doc, here are a few customizations that you should be aware of:
+
+- ``[BindRequired]``: This attribute forces a Model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: Only supported on action parameters. 
+- ``[FromBody]``: Instructs the model binder to use the JSON media type formatter rather than regular model binding.
```

Comment:
> Instructs the model binder to use the JSON media type formatter rather than regular model binding.

Use the configured formatters to bind data from the request body. The formatter is selected based on HTTP content negotiation.